espGarden/main/include/http_server.h

```diff
@@ -4,157 +4,81 @@
-    bool status_flag;        // A flag to indicate some status
-} my_data_type;
-
-esp_err_t set_cors_headers(httpd_req_t *req) {
-    httpd_resp_set_hdr(req, "Access-Control-Allow-Origin", "*");
-    httpd_resp_set_hdr(req, "Access-Control-Allow-Methods", "GET, POST, OPTIONS");
-    httpd_resp_set_hdr(req, "Access-Control-Allow-Headers", "Content-Type");
-    return ESP_OK;
-}
- 
-
-/* Handler function for GET requests */
-esp_err_t get_handler(httpd_req_t *req)
-{
-    set_cors_headers(req);    
-    const char resp[] = "URI GET Response";
-    httpd_resp_send(req, resp, HTTPD_RESP_USE_STRLEN);
-    return ESP_OK;
-}
-
-/* Handler function for POST requests */
-esp_err_t post_handler(httpd_req_t *req)
-{
-    set_cors_headers(req);    
-    char content[100];
-    size_t recv_size = MIN(req->content_len, sizeof(content));
-    int ret = httpd_req_recv(req, content, recv_size);
-    if (ret <= 0) {
-        if (ret == HTTPD_SOCK_ERR_TIMEOUT) {
-            httpd_resp_send_408(req);
-        }
-        return ESP_FAIL;
-    }
-
-    const char resp[] = "URI POST Response";
-    httpd_resp_send(req, resp, HTTPD_RESP_USE_STRLEN);
-    return ESP_OK;
-}
-
-/* Custom function to free context */
-void free_ctx_func(void *ctx)
-{
-    free(ctx);
-}
+/* Handler function for receiving commands */
+esp_err_t rx_handler(httpd_req_t *req) {
+    char rx_buf[RX_TX_BUFF_LEN]; // buffer for receiving data
+    
+    // Receive the data
+    int rx = httpd_req_recv(req, rx_buf, MIN(req->content_len, sizeof(rx_buf) - 1));
 
-/* Handler function for a POST request that uses session context */
-esp_err_t adder_post_handler(httpd_req_t *req)
-{
-    set_cors_headers(req);    
-
-    if (!req->sess_ctx) {
-        req->sess_ctx = malloc(sizeof(my_data_type));
-        req->free_ctx = free_ctx_func;
-    }
-
-    my_data_type *ctx_data = (my_data_type*)req->sess_ctx;
-
-    // Increment request count
-    ctx_data->request_count++;
-
-    // Store received data in context buffer
-    size_t recv_size = MIN(req->content_len, sizeof(ctx_data->data_buffer) - 1);
-    int ret = httpd_req_recv(req, ctx_data->data_buffer, recv_size);
-    if (ret <= 0) {
-        if (ret == HTTPD_SOCK_ERR_TIMEOUT) {
+    if (rx <= 0) {
+        if (rx == HTTPD_SOCK_ERR_TIMEOUT) {
             httpd_resp_send_408(req);
         }
         return ESP_FAIL;
     }
-    ctx_data->data_buffer[recv_size] = '\0'; // Null-terminate the received data
 
-    // Update the status flag (example logic)
-    ctx_data->status_flag = true;
+    rx_buf[rx] = 0;
+    ESP_LOGI(HTTP_LOG_TAG, "Received: %s", rx_buf);
+    
+    // Process the command here (e.g., water the garden for an hour)
+    // process_watering_command(system_state, rx_buf);
 
-    /* Respond */
-    const char resp[] = "Adder POST Response";
+    // Send a receive packet
+    const char resp[] = "rec";
     httpd_resp_send(req, resp, HTTPD_RESP_USE_STRLEN);
+    ESP_LOGI(HTTP_LOG_TAG, "Sent Receipt Confimation: %s", resp);
```

Comment:
HTTP_LOG_TAG and is coming from heltec_unofficial don't we need to declare it as extern in this file

Reply:
See other comments about static variables